Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.cpp

```diff
@@ -0,0 +1,217 @@
+                // onto something which does not understand "asset browser entry" but does understand URLs.
+                QList<QUrl> urls;
+
+                for (const AssetBrowserEntry* entry : entries)
+                {
+                    if (entry)
+                    {
+                        // entries may have aliased paths:
+                        AZStd::string entryPath = entry->GetFullPath();
+                        if (AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetInstance())
+                        {
+                            AZ::IO::FixedMaxPath fixedPath;
+                            if (fileIO->ResolvePath(fixedPath, entryPath.c_str()))
+                            {
+                                entryPath = fixedPath.LexicallyNormal().String();
+                            }
+                        }
+
+                        urls.push_back(QUrl::fromLocalFile(QString::fromUtf8(entryPath.c_str())));
+                    }
+                }
+                // Saving the buffer of asset browser entries back to the mime data
+                QByteArray byteArray(outputString.c_str(), static_cast<int>(outputString.size()));
+                mimeData->setData(AssetBrowserEntry::GetMimeType(), byteArray);
+                mimeData->setData("text/plain", byteArray);
+                mimeData->setUrls(urls);
+            }
+
+            AZStd::string ToString(const AZStd::vector<const AssetBrowserEntry*>& entries)
+            {
+                if (entries.empty())
+                {
+                    return AZStd::string();
+                }
+
+                AZStd::string outputString;
+                outputString.reserve(entries.size() * 64); // roughly average size of uuid or assetids.
+
+                // note that due to the way Qt and mimedata works, its quite possible for duplicates
+                // to be in the entry list.  Intead of requiring every single caller de-duplicate, we will
+                // deduplicate here, in one place.
+                AZStd::unordered_set<const AssetBrowserEntry*> alreadyAdded;
+
+                for (const AssetBrowserEntry* entry : entries)
+                {
+                    if (!entry)
+                    {
+                        continue;
+                    }
+
+                    if (alreadyAdded.find(entry) != alreadyAdded.end())
+                    {
+                        continue;
+                    }
+
+                    AZStd::string outString = WriteEntryToString(entry);
+                    if (outString.empty())
+                    {
+                        continue;
+                    }
+                    outputString.append(outString);
+                    outputString.append("\n");
+                    alreadyAdded.insert(entry);
+                }
+                return outputString;
+            }
+
+            bool FromString(AZStd::string_view inputString, AZStd::vector<const AssetBrowserEntry*>& entries)
```

Comment:
True, or I could make it void.  But as a return type, you wouldn't be able to call this function to decode additional strings into the same vector.  Returning a vector has the semantic of "you get a fresh vector each time, and assigning it clears it".  Feeding in a vector has the semantics of, "add anything you can find, into this vector, do not clear it".

I think the latter use is more flexible in this case.  Especially as we do more and more stuff with copy and paste and multiselect.

Reply:
I'm going to leave it as a bool but fix the logic so that it returns whether it added any new entries to the vector - not whether the vector is empty.